enhancements/cluster-logging/fluentd-tuning.md

```diff
@@ -0,0 +1,150 @@
+
+These settings have weaker forward-compatibility guarantees than the rest of the API.
+In future releases:
+  - `fluentd` settings may be ignored, if the logging implementation changes.
+  - valuesmay be overridden by (present or future) settings in other logging APIs.
+  - new sections may be introduced for new implementations, they may not have equivalent settings to `fluentd`
+
+The `clusterloggings.logging.openshift.io` API will have a new field
+`spec.forwarding.fluentd` map. This is intended to be symmetric with the
+`spec.forwarding.collection` field. The `collection` field is for fluentd
+`input` plugins, the `forwarding` field is for `output` plugins. This emphasizes
+the distinct roles, which may be implemented using different technologies in
+future.
+
+## Motivation
+
+### Goals
+
+Ultimately we want great performance "out of the box" with no user
+intervention. However, today we can't always predict/detect the best settings;
+customers have had to adjust `fluentd` parameters to get good performance.
+
+Furthermore, some requirements *can't* be predicted or detected.  Optimizing for
+low latency often conflicts with optimizing for high throughput. We can't
+predict or detect that a user wants to aggressively optimize for one or the
+other.
+
+Goals:
+* Expose selected fluentd performance optimization parameters in the ClusterLogging API.
+* Apply defaults from forwarding.fluentd to the underlying `fluentd` configuration.
+
+We will be successful when:
+* An administrator is able to set defaults for selected `fluentd` parameters.
+
+### Non-Goals
+
+* Not implying that logging will always use `fluentd` or use it in the same way. `forwarding.fluentd` applies to the current implementation, it may be ignored in future releases.
+* Not allowing general `fluentd` configuration: we expose only a subset of performance tuning parameters. Users that want full control of `fluentd` need to go unmanaged.
+* Not a generic performance tuning API: we may introduce such APIs in future, or introduce new `forwarding` sections. 
+
+## Proposal
+
+Field names are based on fluentd 1.0 `<buffer>` configuration section.
+See the [fluentd documentation](https://docs.fluentd.org/configuration/buffer-section#buffering-parameters) for details.
+
+Example below shows just the `forwarding` section of the ClusterLogging CR.
+
+The field names are copied exactly from `fluentd` 1.0, including the use of "_"
+rather than camelCase. This emphasizes that these settings are not a general
+API, they set defaults directly for the underlying `fluentd`.
+
+Note: two important fields were renamed by `fluentd` between versions 0.12 and 1.0:
+* `queue_limit_length` -> `total_limit_size`
+* `num_threads` -> `flush_thread_count`
+
+```yaml
+forwarding:
+  fluentd:
+    buffer:
+      # Memory use
+      chunk_limit_size: 1m
+      total_limit_size: 32m             # Replaces 'queue_limit_length' in 0.12
+      overflow_action: exception
```

Comment:
Shouldn't that be `throw_exception`?

Reply:
Yep. It used to be "exception", but it was changed, its "throw_exception" in 1.0. Good catch.

Reply:
I would stick to memory, since we're already scraping the logs from a file, which effectively serves as a file buffer.  We can add file buffer if there's a need.

Reply:
Will put together some examples...